### ISSUE TITLE
Suggestion for 2.0: include ClojureScript/D3.js option for graphics / data visualizations/

### ISSUE THREAD

**Issue by michaelochurch** [incanter/incanter#193]
Is there any interest in this? I could start the work on writing this, and get some basic functionality together (e.g. function plotting, scatterplots) pretty quickly?

I think that for heavy-duty uses (e.g. thousands to millions of data points) it's hard to beat interactive browser-based utilities, and IMO it'd be best to use D3 which is already really good at those sorts of things (but the Clojure/ClojureScript wrappers take some of the wonkiness of the language out of the equation.)

**Comment by joinr** [comment]
Resurrecting this to propose Vega plus javafx webengine as an alternative to offer chart....nightcode goes this route to great effect,  vega already exists.

**Comment by jakubholynet** [comment]
You mean https://vega.github.io/vega/ I guess?
søn. 4. feb. 2018 kl. 23:26 skrev Tom <notifications@github.com>:

**Comment by joinr** [comment]
Indeed.   That's what gorilla repl uses for charting (and gyptis, and I think vizard).   I used it for some cljs experiments.  It seems like creating Vega specs from the current set of charts would be straightforward.   Then, users could render to svg, dog (via batik), or to an interactive javafx webengine displaying the chart component (similar to gorilla repl).  That may open up options for other charting backend.  Of course,  writing a portable charting layer in cljc on top of quil is another idea.  Seems like wrapping Vega would be an easy path though.

**Comment by MawiraIke** [comment]
Any update on this issue?

**Comment by mikera** [comment]
This might be interesting: https://github.com/appliedsciencestudio/darkstar
